### ISSUE TITLE
Problem with running google tests in CppUTests

### ISSUE THREAD

**Issue by soltysil** [cpputest/cpputest#1106]
I am trying to run google tests under CppUTests framework, I followed the description in the manual, but it apparently fails to compile.

Error: <path>/cpputest-3.8-installed/include/CppUTest/UtestMacros.h:58:0: error: "TEST" redefined

AllTests.cpp
```
#include "CppUTestExt/GTestConvertor.h"
#include "CppUTest/CommandLineTestRunner.h"

int main(int argc, char** argv)
{
    GTestConvertor convertor;
    convertor.addAllGTestToTestRegistry();
    return CommandLineTestRunner::RunAllTests(argc, argv);
}
```
Makefile
```
INCLUDE_DIRS =../cpputest-3.8-installed/include \
			   ../gmock-1.7.0/include \
			   ../gmock-1.7.0/gtest/include \
			   ../gmock-1.7.0/gtest \

CPPUTEST_CXXFLAGS = -Wno-error=reorder -Wno-error=sign-conversion -Wno-c++14-compat -DCPPUTEST_USE_REAL_GTEST 
```

**Comment by basvodde** [comment]
Could you change the #include order?

**Comment by soltysil** [comment]
When I change #include it complains about TEST and FAIL redefinition:

`In file included from  ../cpputest-3.8-installed/include/CppUTestExt/GTest.h:35:0,
                 from .../cpputest-3.8-installed/include/CppUTestExt/GTestConvertor.h:95,
                 from AllTests.cpp:3:
../gmock-1.7.0/gtest/include/gtest/gtest.h:1877:0: error: "FAIL" redefined [-Werror]
 # define FAIL() GTEST_FAIL()
 ^
In file included from ../cpputest-3.8-installed/include/CppUTest/TestHarness.h:40:0,
                 from ../cpputest-3.8-installed/include/CppUTest/CommandLineTestRunner.h:31,
                 from AllTests.cpp:1:
../cpputest-3.8-installed/include/CppUTest/UtestMacros.h:296:0: note: this is the location of the previous definition
 #define FAIL(text)\
`

**Comment by basvodde** [comment]
Uh. And can you check the file tests/CppUTestExt/AllTest.cpp and see if that compiles? That is the self-test main for GTestRunner.

**Comment by soltysil** [comment]
It compiled with few warnings. In order to make It work I had to set CPPUTEST_WARNINGFLAGS in my makefile. This bypasses setting default warning flags in MakefileWorker.am.

But then I noticed that MakefileWorker.am already has a logic to rest CPPUTEST_WARNINGFLAGS when GMOCK_HOME is defined. The check doesn't seem to be correct, as event though I have GMOCK_HOME exported, the condition never gets evaluated to true.

I guess the intention in MakefileWorker.am was to do this:

`ifdef GMOCK_HOME
...
endif
`
rather than this:
`ifdef $(GMOCK_HOME)
...
endif
`
